mobile/src/main/java/org/openhab/habdroid/util/ColorPicker.kt

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.habdroid.util
+
+import android.content.Context
+import android.graphics.Color
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ArrayAdapter
+import android.widget.GridView
+import android.widget.LinearLayout
+import androidx.annotation.ColorInt
+import androidx.appcompat.app.AlertDialog
+import com.google.android.material.slider.Slider
+import org.openhab.habdroid.R
+import org.openhab.habdroid.model.Item
+import kotlin.math.max
+
+class ColorPicker(val context: Context, val inflater: LayoutInflater) : Slider.LabelFormatter,
+    Slider.OnSliderTouchListener {
+    private var listener: OnColorPickerChangeListener? = null
+    private var slider: Slider? = null
+
+    fun showDialog(
+        colors: IntArray,
+        listener: OnColorPickerChangeListener,
+        item: Item
+    ): AlertDialog {
+        val alertLayout: View = inflater.inflate(R.layout.color_picker_dialog, null)
+
+        val gridItems = ArrayList<ColorPickerItem>()
+        colors.forEach { color ->
+            gridItems.add(ColorPickerItem(color))
+        }
+
+        this.listener = listener
+
+        val colorGridAdapter = ColorPickerAdapter(context, R.layout.color_picker_item, gridItems, listener)
+        val gridView: GridView = alertLayout.findViewById(R.id.grid)
+        gridView.adapter = colorGridAdapter
+        val resources = context.resources
+        val columnCount = resources.pixelToDp(resources.displayMetrics.widthPixels).toInt() / 100
+        gridView.numColumns = max(columnCount, 2)
```

Comment:
This is a workaround. If I use `AUTO_FIT`, the gridview only has two columns, which is too less.